pkg/deployer/baremetal/deployer.go

```diff
@@ -285,24 +287,34 @@ func (d *Deployer) ListGreptimeDBClusters(ctx context.Context, options *ListGrep
 }
 
 func (d *Deployer) CreateGreptimeDBCluster(ctx context.Context, clusterName string, options *CreateGreptimeDBClusterOptions) error {
-	if err := d.am.PrepareArtifact(ctx, GreptimeArtifactType, d.config.Cluster.Artifact); err != nil {
-		return err
-	}
+	var binPath string
+	if d.config.Cluster.Artifact != nil {
+		if d.config.Cluster.Artifact.Local != "" {
+			binPath = d.config.Cluster.Artifact.Local
+		} else {
+			src, err := d.am.NewSource(artifacts.GreptimeBinName, d.config.Cluster.Artifact.Version, artifacts.ArtifactTypeBinary, false)
+			if err != nil {
+				return err
+			}
 
-	binary, err := d.am.BinaryPath(GreptimeArtifactType, d.config.Cluster.Artifact)
-	if err != nil {
-		return err
+			destDir := filepath.Join(d.baseDir, "artifacts", "binaries", artifacts.GreptimeBinName, d.config.Cluster.Artifact.Version, "pkg")
```

Comment:
we are still using literal strings here, but later we can abstract them into one cluster-dir-manager for bare-metal.

i will work on this.